addons/project/security/project_security.xml

```diff
@@ -228,6 +228,15 @@
         <field name="groups" eval="[(4,ref('base.group_user'))]"/>
     </record>
 
+    <record model="ir.rule" id="report_project_task_user_rule">
+        <field name="name">Tasks Analysis: project visibility</field>
+        <field name="model_id" ref="model_report_project_task_user"/>
+        <field name="domain_force">['|',
+            ('project_id.privacy_visibility', '!=', 'followers'),
+            ('project_id.message_partner_ids', 'in', [user.partner_id.id])]
+        </field>
+    </record>
```

Comment:
So, you filter the data even is the user is a project manager ? 🤔 
It seems weird.